### PR TITLE
feat: allow dynamic async/rejectable allowedFlows COMPASS-6811

### PR DESCRIPTION
**Pull request by addaleax**
This should enable us to display a modal in Compass when re-authentication is required after the initial auth, and prompt the user whether they want to reauthenticate or just return to the connection screen again.

### CODE REVIEW

File: src/plugin.spec.ts [hunk 1]
```diff
@@ -241,7 +241,10 @@ describe('OIDC plugin (local OIDC provider)', function () {
         expect(timeouts).to.have.lengthOf(1);
         expect(timeouts[0].refed).to.equal(false);
         expect(timeouts[0].cleared).to.equal(false);
-        expect(timeouts[0].timeout).to.equal(9_700_000);
+        // openid-client bases expiration time on the actual current time, so
+        // allow for a small margin of error
+        expect(timeouts[0].timeout).to.be.greaterThanOrEqual(9_600_000);
+        expect(timeouts[0].timeout).to.be.lessThanOrEqual(9_800_000);
```
Comment (addaleax) [inline src/plugin.spec.ts:247]: (This is unrelated, the test just failed on an earlier version of this PR)

**Review by gribnoysup — APPROVED** [review]
